### PR TITLE
Users can register an account #81

### PR DESCRIPTION
**Pull request by aaronkisac**

### CODE REVIEW

File: client/src/helpers/api.js [hunk 1]
```diff
@@ -117,3 +117,16 @@ export const deleteTopic = async topic_id => {
 export const addTopic = async (title, lesson_id) => {
   return await instance.post("/api/topics", { title, lesson_id });
 };
+
+export const getUsers = async () => {
+  return await instance.get("/api/users").then(res => res.data);
```
Comment (islam95) [inline client/src/helpers/api.js]: I think you don't need to use `.then` when you use `async/await`.

File: server/api/index.js [hunk 2]
```diff
@@ -191,4 +191,31 @@ router.post("/topics", async (req, res) => {
   }
 });
 
+// // Add User
+router.post("/users", async (req, res) => {
+  const { name, email, password } = req.body;
+  if (
+    (await db.checkUserByNameExist(name)) === false &&
```
Comment (islam95) [inline server/api/index.js]: Here, you are not allowing new users to have the same name. That is not a good idea. The name is not unique, email is.

Comment (aaronkisac) [reply server/api/index.js]: You are right I will change.

File: client/src/components/Register/Register.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      email: "",
+      password: "",
+      users: [],
+      message: "",
+      messageAlert: ""
+    };
+  }
+  componentDidMount = async () => {
+    const res = await getUsers();
+    const data = res;
```
Comment (islam95) [inline client/src/components/Register/Register.js]: Line 18 is redundant. Why create unnecessary variable if they are the same. You can rename `res` to `users` and then setState with just `{ users }`, it's the same as `{users: users}`.

File: client/src/helpers/api.js (hunk 1 above)
Comment (aaronkisac) [reply client/src/helpers/api.js]: ok but How to take res.data I don't know

File: client/src/components/Register/Register.js (hunk 3 above)
Comment (aaronkisac) [reply client/src/components/Register/Register.js]: you are right again. I didnt see that. I did this at night. I will change that

File: client/src/helpers/api.js (hunk 1 above)
Comment (islam95) [reply client/src/helpers/api.js]: You can say `const res = await instance.get("...")`, and in next line `return res.data`. However, I don't think you need `res.data` in this function. Test it inside your component by logging console and see what it shows with and without `res.data`. Also, in component you can do the same when you `await` you might need `res.data`. You can have a look.

File: server/api/index.js (hunk 2 above)
Comment (islam95) [reply server/api/index.js]: 👍

File: client/src/components/Register/Register.js (hunk 3 above)
Comment (islam95) [reply client/src/components/Register/Register.js]: 👍

File: client/src/helpers/api.js (hunk 1 above)
Comment (aaronkisac) [reply client/src/helpers/api.js]: I need res.data. Loic told me about this if you need res.data you can do in api.js this is good coding. he said.

Comment (islam95) [reply client/src/helpers/api.js]: Have you tried what I said? If not, fair enough.

Comment (aaronkisac) [reply client/src/helpers/api.js]: I looked again I don't need getUsers . I cleaned that it is still working.

**Review by llh1 — APPROVED** [review]

**Pull request merged**